File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d = {}
        for i in nums:
            if i in d:
                d[i]+=1
            else:
                d[i] = 1
        heap = []
        for item in d.items():
            heapq.heappush(heap, [item[1], item[0]])
            if(len(heap) > k):
                heapq.heappop(heap)
        return [item[1] for item in heap]

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer, Integer> freq = new HashMap<>();
        for(int i = 0;i<nums.length;i++){
            freq.put(nums[i], freq.getOrDefault(nums[i],0)+1);
        }
        PriorityQueue<Map.Entry<Integer,Integer>> pq = new PriorityQueue<>((a,b) -> {return b.getValue() - a.getValue();});
        for(Map.Entry<Integer,Integer> item : freq.entrySet()){
            pq.add(item);
        }
        int [] res = new int[k];
        int i = 0;
        while(i < k){
            res[i] = pq.poll().getKey();
            i++;
        }
        return res;
    }
}